### PR TITLE
mldsa: fix centeredAbs at (q-1)/2 and add the CCTV accumulated tests

### PR DESCRIPTION
**Pull request by filippo-claude**
## Summary

Two related changes:

1. **Bugfix:** `centeredAbs(a)` is off by one at `a = (q-1)/2`. q is odd,
   so per FIPS 204 §2.3 the centered representative of `(q-1)/2 = 4 190 208`
   is itself, not `q − a = 4 190 209`. The implementation used the
   inclusive predicate `(q-1)/2 <= a` to pick the `q − a` branch, and
   `TestCenteredAbs`'s expected-value computation has the same off-by-one
   so the test passed by being wrong in the same way as the implementation.

2. **New tests:** apply the [CCTV ML-DSA accumulated test
   vectors](https://github.com/C2SP/CCTV/tree/main/ML-DSA/accumulated).
   These capture randomized ML-DSA end-to-end results and exhaustive
   field-operation results, covering all the field edge cases.

## Impact of the bugfix

`centeredAbs` is consumed by `centeredMax` and from there by `infinityNorm`.
All four `infinityNorm` call sites (in `signInternalWithMu` and
`verifyInternalWithMu`) bound the input coefficients well below
`(q-1)/2`: `2^log2γ₁ + β`, `γ₂`, `τ·2^(d-1)`. So the off-by-one never
fires on honest inputs and **no signing/verifying path is functionally
affected**. The fix is for spec conformance and to remove a sharp edge
in `infinityNorm` if it's ever applied to a polynomial with arbitrary
coefficients in the future.

## Tests added

* `TestCCTVAccumulated` — accumulated random key-generation +
  deterministic signature vectors at 100 and 10 000 iterations for each
  of ML-DSA-44, -65, -87. The 10 000-iteration vectors are skipped in
  `-short` mode. Total runtime ~50 s without `-short`, ~1 s with it.
* `TestCCTVFieldOps` — the exhaustive field-operations vector,
  iterating every element of ℤq (8 380 417 inputs) and exercising
  `centeredAbs`, `centeredMax`, `infinityNorm`, `power2Round`,
  `decompose`, and `useHint` at every coefficient. This is the test
  that surfaced the `centeredAbs` bug above; it would also have caught
  the `useHint` r0 = 0 bug fixed in #48 (60 disagreeing inputs across
  the two γ₂ values), since `algebra_test.go`'s `TestUseHint`
  re-implemented the same off-by-one bug as the production code,
  whereas this test compares against an externally-derived hash.

## Test plan

- [x] `go test -count=1 ./internal/signature/mldsa/ ./signature/mldsa/`
- [x] `go test -short -count=1 ./internal/signature/mldsa/`
- [x] `TestCCTVFieldOps` produces
      `f930663417278156ab05d940294a77210a809c924d8ab63ec72f4526247602c7`
      (matches the published hash) only with the centeredAbs fix applied.
- [x] All six `TestCCTVAccumulated` vectors at 100 and 10 000 iterations
      match the published hashes for ML-DSA-44, -65 and -87.
- [x] Reverting the #48 useHint fix on top of this branch makes
      `TestCCTVFieldOps` fail (alongside the existing `algebra_test.go`'s
      now-corrected `TestUseHint`).

## References

- CCTV README: https://github.com/C2SP/CCTV/blob/main/ML-DSA/accumulated/README.md
- Sibling fix in this same area on tink-go main: #48 (UseHint at r0 = 0)

### CODE REVIEW

**Comment by FiloSottile** [comment]
I kicked Opus 4.7 off with this prompt

> apply the https://github.com/C2SP/CCTV/tree/main/ML-DSA/accumulated tests to https://github.com/tink-crypto/tink-go/releases/tag/v2.6.0
>
> if you find a bug, triage it and write a report on it/them

intentionally telling it to target v2.6.0 to see if it would find #48, as a test of both the cloud Claude Code and the accumulated test.

It immediately found both that and the centeredAbs bug. Then it struggled a bit with reproducing the hash, so I told it

> You seem to be struggling with the accumulated test format. https://github.com/FiloSottile/mldsa-py/commit/4c650a4a9db751a179cf2ec57d5fe01f2bee191c is an example. What's a diff to the CCTV README that would have clarified the format for you? Maybe it should include an example for a q value?

which led to this improvement https://github.com/C2SP/CCTV/commit/67c1397af2a57f935cc96ee112b446c051cdb68a (my spec was wrong! I was hashing r0 mod q, not r0).

I then told it to check if the bugs were fixed on main, and to open a PR for the one that wasn't, and I reviewed and gave feedback on the code and commits until I was happy with them. (So this is not a drive-by vibe-coded PR!)

**Comment by lukaszobernig** [comment]
@FiloSottile, ahh nice, thanks also for adding the test vectors! I will import and merge.

File: internal/signature/mldsa/algebra.go [hunk 1]
```diff
@@ -194,11 +194,21 @@ func (a rZq) useHint(gamma2 uint32, h rZq) rZq {
...
+	// FIPS 204 §2.3 defines a mod± α as the unique m' ∈ ℤ with
+	//   m' ≡ m (mod α) and -⌈α/2⌉ < m' ≤ ⌊α/2⌋.
+	// q = 8380417 is odd, so ⌊q/2⌋ = (q-1)/2 = 4190208 and ⌈q/2⌉ = (q+1)/2 =
+	// 4190209, giving -4190208 ≤ m' ≤ 4190208, i.e. m' ∈ [-(q-1)/2, (q-1)/2].
+	// For a ∈ [0, q):
+	//   a ≤ ⌊q/2⌋ = (q-1)/2  ⟹  m' = a, |m'| = a
+	//   a > ⌊q/2⌋ = (q-1)/2  ⟹  m' = a − q, |m'| = q − a
+	// So the predicate for the q − a branch is a > (q-1)/2; the boundary
+	// value a = (q-1)/2 is non-negative and stays in the +a branch.
+	c := subtle.ConstantTimeLessOrEq((q-1)/2+1, int(a))
```
Comment (gendx) [inline internal/signature/mldsa/algebra.go:211]: This can be simplified to `(q+1)/2`.

**Comment by lukaszobernig** [comment]
Merged in https://github.com/tink-crypto/tink-go/commit/24bcd560e1c6a786193179678a2fcb55ab73ba0b. I had to take out the 10000 iteration tests since they timed out in some of our CI runners.
